### PR TITLE
Fixed NullPointerException in the hashcode() method (Student)

### PR DESCRIPTION
**Pull request by kmusienko**
When we try to add new student, we get NullPointerException in the field Group in Student entity, because Group field hasn't been set yet.

### CODE REVIEW

File: src/main/java/com/softserve/teamproject/entity/Student.java [hunk 1]
```diff
@@ -111,7 +111,7 @@ public boolean equals(Object otherObject) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(firstName, lastName, group.getId());
+    return Objects.hash(firstName, lastName);
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: if you still want to use group, you can make like (group!=null?group.getName:null)

Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: but it's up to you, if not needed you can merge

Comment (kmusienko) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: what? in general it's bad idea to use in hashcode() any other entities.

Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: well, if we have equals and hashcode of name and lastname of student, then two students with same names but in different groups will be equal, and you can't you HashSet etc.

Comment (kmusienko) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: yes, in order to solve this problem we have to have some business key, for example 'studentNickname' which would be unique. but using other related entities in hashcode() doesn't solve the problem at all.

Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: we don't have unqiue key now, so

Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: business key is a combination of firstname & lastname & groupid

Comment (kmusienko) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: it's bad practise to use related entities in hashcode anyway.

Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/entity/Student.java]: hmm i am wrong combination of firstname & lastname & groupid isnt business key,  we can have 2 students with same firstname and lastname in the same group, such a coincidence is a possible ;D

**Pull request merged**